### PR TITLE
Grease pencil node Mask layer

### PR DESCRIPTION
**Pull request by sedrew**
image
ffdd55s

### CODE REVIEW

**Comment by 3DSinghVFX** [comment]
Thanks, It is a very nice addition. I'll review this shortly.

File: animation_nodes/nodes/gpencil/gp_object_input.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def executeSingle(self, object, useWorldSpace, layerName):
         evaluatedObject = getEvaluatedID(object)
         layer = self.getLayer(evaluatedObject, layerName)
         return GPLayer(layer.info, self.getFrames(layer, evaluatedObject, useWorldSpace), layer.blend_mode,
-                       layer.opacity, layer.pass_index)
+                       layer.opacity, layer.pass_index, layer.maskLayer)
```
Comment (3DSinghVFX) [inline animation_nodes/nodes/gpencil/gp_object_input.py]: Change this `layer.maskLayer` --> `layer.mask_layer`.

File: animation_nodes/nodes/gpencil/gp_object_input.py [hunk 2]
```diff
@@ -54,7 +54,7 @@ def executeList(self, object, useWorldSpace):
         gpencilLayers = []
         for layer in self.getLayers(evaluatedObject):
             gpencilLayers.append(GPLayer(layer.info, self.getFrames(layer, evaluatedObject, useWorldSpace),
-                                         layer.blend_mode, layer.opacity, layer.pass_index))
+                                         layer.blend_mode, layer.opacity, layer.pass_index, layer.maskLayer))
```
Comment (3DSinghVFX) [inline animation_nodes/nodes/gpencil/gp_object_input.py]: Here also change this `layer.maskLayer` --> `layer.mask_layer`.

**Review by 3DSinghVFX — COMMENTED** [review]
Make the following changes:

**Comment by sedrew** [comment]
I changed this

File: animation_nodes/ui/node_menu.py [hunk 3]
```diff
@@ -510,6 +510,7 @@ def draw(self, context):
         insertNode(layout, "an_GPSetLayerOpacityNode", "Set Layer Opacity")
         insertNode(layout, "an_GPSetLayerPassIndexNode", "Set Layer Pass Index")
         insertNode(layout, "an_GPTransformLayerNode", "Transform Layer")
+        insertNode(layout, "an_GPLayerMaskNode", "Layer Mask")
```
Comment (3DSinghVFX) [inline animation_nodes/ui/node_menu.py:512]: Move this line above the `Transform Layer` line.

**Review by 3DSinghVFX — COMMENTED** [review]
This looks good to me. I have a concern with the name of the `GP Layer Mask` node. So, I'll discuss it with Omar and let you know.

**Comment by sedrew** [comment]
I made the last changes. I'm thinking of really changing the name to  `GP Set Layer Mask` i will wait for an answer from Omar

**Comment by 3DSinghVFX** [comment]
I also thought that name too but we have to discuss it with Omar first :)

**Comment by 3DSinghVFX** [comment]
Hi...
I discussed it with Omar. Please create the pull request directly to Animation Nodes (https://github.com/JacquesLucke/animation_nodes/pulls), I'll review it there.

- Update the master.
- Create a separate branch from the master, and make the changes then create the pull request.

Thanks :)

**Comment by sedrew** [comment]
Okey, I saw you made changed in code. I should expect to merge your pull request with master https://github.com/JacquesLucke/animation_nodes/pull/1307? and then update my master.

**Comment by 3DSinghVFX** [comment]
Yes, that is a good idea. Because there are some files which are common and have some changes.

**Comment by sedrew** [comment]
Okey, I will be wait.

**Comment by 3DSinghVFX** [comment]
Thanks :)

**Comment by 3DSinghVFX** [comment]
Hi...
https://github.com/JacquesLucke/animation_nodes/pull/1307 is merged into the master.

**Comment by sedrew** [comment]
Thanks, I'll do it soon

**Pull request closed without merging**